torch/distributed/_composable_state.py

```diff
@@ -0,0 +1,21 @@
+import torch.nn as nn
+from typing import cast, Dict, Optional
+
+
+class _State:
+    pass
+
+_module_state_mapping: Dict[nn.Module, _State] = {}
+
+def _insert_module_state(module: nn.Module, state: _State) -> None:
+    global _module_state_mapping
+    assert module not in _module_state_mapping
+    _module_state_mapping[module] = state
+
+
+def _get_module_state(module: nn.Module) -> Optional[_State]:
+    global _module_state_mapping
+    if isinstance(module, _State):
```

Comment:
- Why `module` can be a `_State` instance?
- If it's necessary, then do we need to update the type of the `module` arg?

Reply:
This is actually for non-composable API, wrapper-based FSDP. FSDP shares most utils between composable and wrapper-based implementations. The reason we can do this check (specifically for wrapper-based FSDP), is that we add `class FullyShardedDataParallel(nn.Module, _FSDPState):` to ensure `_get_module_state` can correctly detect it. That's also the reason why `_get_module_state` is not inside `_composable/contract.py` -- to avoid circular dependency.